File: service/board-service/src/main/java/jeonghyeon/msa/board/util/PageLimitCalculator.java
package jeonghyeon.msa.board.util;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class PageLimitCalculator {

    public static Long calculatePageLimit(Long page, Long pageSize, Long movablePageCount) {
        return (((page) / movablePageCount) + 1) * pageSize * movablePageCount + 1;
    }

}

File: service/memo-service/src/main/kotlin/jeonghyeon/msa/memo/util/PageLimitCalculator.kt
package jeonghyeon.msa.memo.util

object PageLimitCalculator {
    fun calculatePageLimit(page: Long, pageSize: Long, movablePageCount: Long): Long {
        return (((page) / movablePageCount) + 1) * pageSize * movablePageCount + 1
    }
}